Gems/Atom/RPI/Code/Tests/Image/StreamingImageTests.cpp

```diff
@@ -726,4 +726,41 @@ namespace UnitTest
 
         RPI::ImageSystemInterface::Get()->Update();
     }
+
+    TEST_F(StreamingImageTests, GetSubImagePixelValues)
+    {
+        using namespace AZ;
+
+        Data::Asset<RPI::StreamingImageAsset> imageAsset = BuildTestImage(AZ::RHI::Format::R8_UNORM);
+
+        auto streamingImageAsset = imageAsset.Get();
+        EXPECT_NE(streamingImageAsset, nullptr);
+
+        // Validate retrieving one pixel at a time
+        auto size = streamingImageAsset->GetImageDescriptor().m_size;
+        for (uint32_t y = 0; y < size.m_height; ++y)
+        {
+            for (uint32_t x = 0; x < size.m_width; ++x)
+            {
+                auto pixelDataValue = imageAsset->GetSubImagePixelValue<float>(x, y);
+                auto pixelExpectedValue = static_cast<uint8_t>(y * size.m_width + x) / static_cast<float>(std::numeric_limits<AZ::u8>::max());
+
+                EXPECT_TRUE(AZ::IsClose(pixelDataValue, pixelExpectedValue));
+            }
+        }
+
+        // Validate retrieving a region of pixels
+        AZStd::vector<float> pixelValues(size.m_width * size.m_height);
+        auto topLeft = AZStd::make_pair<uint32_t, uint32_t>(0, 0);
+        auto bottomRight = AZStd::make_pair<uint32_t, uint32_t>(size.m_width - 1, size.m_height - 1);
+        AZStd::span<float> valueSpan(pixelValues.begin(), pixelValues.size());
+        streamingImageAsset->GetSubImagePixelValues(topLeft, bottomRight, valueSpan);
```

Comment:
Nit: you should be able to just call GetSubImagePixelValues(topLeft, bottomRight, pixelValues). You shouldn't need to explicitly make a span<>.

Reply:
I thought I got a compile error about it being ambiguous when I tried that, let me give it another shot

Reply:
Nevermind, it did work, have updated :)